geode-assembly/src/acceptanceTest/resources/org/apache/geode/cache/wan/docker-compose.yml

```diff
@@ -17,12 +17,8 @@
 version: '3'
 services:
   locator:
-    container_name: 'locator'
     image: 'geode:develop'
     hostname: locator
-    expose:
```

Comment:
I see you've removed all the `expose` directives in the various `docker-compose.yml` files. I suppose these still work. I guess that's because your Java code is calling testcontainers API to expose the pertinent ports (to other containers).

On the one hand I can see why perhaps this information could be deemed redundant (since it's also in the Java code). On the other hand, if I could have the exposed ports information in only one place, I'd prefer it in the `yml` files since these can be used from the terminal to spin up services for experimentation.

Not only is it useful to have them in the `yml` file for experimentation, their presence there gives us a clear picture (at a glance) of all the exposed ports. That's useful for maintainers.

Would it make sense to reinstate these `expose` directives? Could we rely on them here in the `yml` files and remove the redundant Java code?

Reply:
Totally agree, but I couldn't get _testcontainers_ to work properly with the `expose` directives left in place. Possibly I'm doing something wrong. It does do something a bit weird in that it spins up an _ambassador_ container which proxies the connections to the actual containers. I think that may be part of the need to not expose the ports, but I'm not sure. More details here: https://www.testcontainers.org/modules/docker_compose/#example

Reply:
Apparently `expose` (in a docker compose file) isn't really needed:

https://stackoverflow.com/questions/65784039/why-a-service-configured-with-only-expose-is-able-to-communicate-with-internet/65785558#65785558

At best is was mere documentation. So I'm ok w/ removing these.